### PR TITLE
Preference screen size

### PR DESCRIPTION
**Pull request by ppwadhwa**
# Description
Adds in a checkbox to save the window geometry.  Can be reviewed after 2211 is merged.

## Type of change

- [ ] Bug-fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
Screen Shot 2021-03-12 at 11 16 29 AM

### CODE REVIEW

**Comment by ppwadhwa** [comment]
@sofroniewn @tlambert03 @jni @goanpeca 
This PR (adding the screen size checkbox) is ready for review!

**Review by tlambert03 — APPROVED** [review]

File: napari/utils/settings/_defaults.py
Comment (sofroniewn) [inline napari/utils/settings/_defaults.py]: I can't remember now, but I thought @jni had said last week he was fine remembering window geometry by defualt, he just wanted to be able to turn it off if he wanted to?

Personally I've really enjoyed having it be remembered, and I know that was a time when we were thinking we wouldn't even give users control over this as it is so helpful?

Maybe @jni can weigh in before merge?

**Review by sofroniewn — APPROVED** [review]
I'm approving - I love how short a PR this is! It indicates that the whole system is well built :-)

I've asked a question to @jni about the default - I thought we were going to remember the position by default and you just wanted to be able to turn that off to get the experience of a new user (though it still wasn't clear to me why you couldn't just reset to defaults in that case?) I know @liaprins-czi had originally recommended not even giving users control over this, and I think I would have been fine having it always on.

Maybe there was one argument about if you had connected to multiple screens on the previous session then they might get remembered wrong, but I think our code can handle that. Also that use case to me suggests that a user might want a "reset screen position" button, rather than to never save the window position. I could imagine wanting to reset the screen position without wanting to reset all the other settings. For that I might imagine instead of a checkbox a push button which when clicked would immediately reset only the window positions.

Curious what others think about all this, we might want to have a little bit more discussion before merge. I hope that's ok with everyone

File: napari/utils/settings/_defaults.py
Comment (ppwadhwa) [reply napari/utils/settings/_defaults.py]: @goanpeca also mentioned to me that it should probably be set to True as the default, so I went ahead and made this change.

Comment (sofroniewn) [reply napari/utils/settings/_defaults.py]: Great, I'm all for merging this now then as is, and we can decide if we maybe want to push button instead later - but curious what @jni thinks before final decision

**Review by goanpeca — APPROVED** [review]
LGTM, thanks @ppwadhwa for working on this :)

**Review by Carreau — APPROVED** [review]
Apologies for the late review, I had a few days off this week. 

+1 for this; if we want to add other ways to start w/o preferences that can be done separately IMHO.

And if @jni want a new user experience; it's likely not limited to window position; but many other parameters like shortcut, opened panels, language... etc. So I think a flag `--no-pref`, or an environment variable to temporarily change where the setting are loaded from might be a better approach.

**Comment by sofroniewn** [comment]
Ok, thanks for all the input here. It's been about a week since I left my original comments requesting more discussion. Having the checkbox was the original request, so let's just stick with it. We can continue discuss ways to further customize/ turn-off preferences in other places. I will merge now.

**Pull request merged**
